### ISSUE TITLE
Pysimdjson fails to install on python 3.6

### ISSUE THREAD

**Issue by anudeepsamaiya** [TkTech/pysimdjson#56]
```Collecting pysimdjson==2.5.0
  Using cached https://files.pythonhosted.org/packages/9b/f6/c63260f8788574de8fdd0bbe70f803328cb058141c0903ba29637d89f863/pysimdjson-2.5.0.tar.gz
Installing collected packages: pysimdjson
  Running setup.py install for pysimdjson ... error
    Complete output from command /home/ubuntu/ctix-2/venv/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-wzmco2i3/pysimdjson/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-jelehspx/install-record.txt --single-version-externally-managed --compile --install-headers /home/ubuntu/ctix-2/venv/include/site/python3.6/pysimdjson:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.6
    creating build/lib.linux-x86_64-3.6/simdjson
    copying simdjson/__init__.py -> build/lib.linux-x86_64-3.6/simdjson
    running build_ext
    building 'csimdjson' extension
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/simdjson
    x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/home/ubuntu/ctix-2/venv/lib/python3.6/site-packages/pybind11/include -I/home/ubuntu/ctix-2/venv/include -I/usr/include/python3.6m -c simdjson/binding.cpp -o build/temp.linux-x86_64-3.6/simdjson/binding.o -std=c++11
    In file included from /home/ubuntu/ctix-2/venv/lib/python3.6/site-packages/pybind11/include/pybind11/pytypes.h:12:0,
                     from /home/ubuntu/ctix-2/venv/lib/python3.6/site-packages/pybind11/include/pybind11/cast.h:13,
                     from /home/ubuntu/ctix-2/venv/lib/python3.6/site-packages/pybind11/include/pybind11/attr.h:13,
                     from /home/ubuntu/ctix-2/venv/lib/python3.6/site-packages/pybind11/include/pybind11/pybind11.h:44,
                     from simdjson/binding.cpp:5:
    /home/ubuntu/ctix-2/venv/lib/python3.6/site-packages/pybind11/include/pybind11/detail/common.h:112:20: fatal error: Python.h: No such file or directory
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

    ----------------------------------------
Command "/home/ubuntu/ctix-2/venv/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-wzmco2i3/pysimdjson/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-jelehspx/install-record.txt --single-version-externally-managed --compile --install-headers /home/ubuntu/ctix-2/venv/include/site/python3.6/pysimdjson" failed with error code 1 in /tmp/pip-install-wzmco2i3/pysimdjson/
You are using pip version 19.0, however version 20.2.4 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.```

**Comment by anudeepsamaiya** [comment]
fixed it by installing ```sudo apt-get install build-essential libssl-dev libffi-dev python3.6-dev python3-dev python-dev```

**Comment by lemire** [comment]
The "fatal error: Python.h" error is a tell that it is a system configuration issue.  You would hope that `pip` would be smarter and warn you in a nicer way, but it does not.

**Comment by TkTech** [comment]
If you update your pip (as it suggests doing in the last line), you should just get a binary package when you try to install. We produce wheels for 3.6, so there's no reason to build from source unless you really want to.

**Comment by lemire** [comment]
@TkTech 

_If you update your pip (as it suggests doing in the last line), you should just get a binary package when you try to install. We produce wheels for 3.6, so there's no reason to build from source unless you really want to._

It is not your fault in the least, and maybe it is nobody's fault, but pip could easily check whether building a package has a chance to work. And if Python.h is missing, it won't work. It is fine to try and fail, but it should then, I think, display an informative error message instead of this gibberish.

Quality of life and all that.

**Comment by TkTech** [comment]
It's a much harder problem than it sounds, especially while maintaining compatibility with 10 years of tooling. Just to parse the output would require testing against the output of hundreds of compiler/version combos. Including a C/C++ parser would be way out of scope (and what about language X?), and Python.h is often not required. I could have just built a normal DLL and used the built-in ctypes library or CFFI.

**Comment by lemire** [comment]
Let me rephrase the "could easily" part of my comment. I am sure it is super hard.
